### PR TITLE
[#116] Add assertion cases

### PR DESCRIPTION
**Pull request by sudiptob2**
closes chkware/cli#116

### CODE REVIEW

File: chk/modules/assertion/support.py [hunk 1]
```diff
@@ -113,6 +113,13 @@ def case_AssertLessOrEqual(self):
         actual = type_converter(self.actual) if type(self.actual) == str else self.actual
         self.assertLessEqual(actual, self.expect)
 
+    def case_AssertListContains(self):
+        """Asserts expected exist in the actual."""
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+        expect = type_converter(self.expect) if type(self.expect) == str else self.expect
```
Comment (0hsn) [inline chk/modules/assertion/support.py:119]: we don't need to convert `expect` for now. as user can control the value

File: tests/resources/storage/sample_config/bitcoin-usd-testcase.chk [hunk 2]
```diff
@@ -28,6 +28,7 @@ spec:
     - {type: AssertIsMap, actual: {'a': 1, 'b': 2}}
     - {type: AssertIsList, actual: $Response.body.coin.exp}
     - {type: AssertIsList, actual: [1, 1, 2]}
+    - {type: AssertListContains, actual: [1, 1, 2], expected: 1}
```
Comment (0hsn) [inline tests/resources/storage/sample_config/bitcoin-usd-testcase.chk:31]: add a case for json list [sample](https://stackoverflow.com/a/3916173/5571663)

**Review by 0hsn — CHANGES_REQUESTED** [review]

File: tests/resources/storage/sample_config/bitcoin-usd-testcase.chk (hunk 2 above)
Comment (sudiptob2) [reply tests/resources/storage/sample_config/bitcoin-usd-testcase.chk:31]: Thanks, will update

File: chk/modules/assertion/support.py [hunk 3]
```diff
@@ -116,9 +116,8 @@ def case_AssertLessOrEqual(self):
     def case_AssertListContains(self):
         """Asserts expected exist in the actual."""
         actual = type_converter(self.actual) if type(self.actual) == str else self.actual
-        expect = type_converter(self.expect) if type(self.expect) == str else self.expect
         assert type(actual) == list, f"`{self.actual}` is not list"
```
Comment (0hsn) [inline chk/modules/assertion/support.py:119]:
```suggestion
        assert type(actual) == list, f"`{self.actual}` is not a list"
```

Comment (sudiptob2) [reply chk/modules/assertion/support.py:119]: I I also change in line no `85`

Comment (0hsn) [reply chk/modules/assertion/support.py:119]: if you find reasonable, why not?

File: chk/modules/assertion/support.py [hunk 4]
```diff
@@ -96,23 +104,105 @@ def case_AssertCount(self):
...
+        actual = type_converter(self.actual) if type(self.actual) == str else self.actual
+
+        assert type(actual) == dict, f"`{self.actual}` is not a map"
+
+        has_keys, failed_key = False, None
+        for key in self.expect:
+            if key in actual.keys():
+                failed_key, has_keys = key, True
+                break
+        assert not has_keys, f"key `{failed_key}` is present in the map"
```
Comment (0hsn) [inline chk/modules/assertion/support.py:187]:
```suggestion
        assert not set(temp1) & set(temp2), f"at least one of the keys `{self.expect}` is present in the map"
```

**Review by 0hsn — CHANGES_REQUESTED** [review]
Take some more time, look into the loops, and let's write some quality code.

File: chk/modules/assertion/support.py (hunk 4 above)
Comment (sudiptob2) [reply chk/modules/assertion/support.py:187]: sure

Comment (0hsn) [reply chk/modules/assertion/support.py:187]: you can also, capture which key duplicates capturing the result, if you want

Comment (sudiptob2) [reply chk/modules/assertion/support.py:187]: Yes may be using set operation such as intersections

**Review by 0hsn — COMMENTED** [review]
👌

**Review by 0hsn — APPROVED** [review]

**Pull request merged**